### PR TITLE
make response local in fetchData fun

### PR DESCRIPTION
**Pull request by r426**

### CODE REVIEW

File: app/src/main/java/com/ryeslim/currencyexchange/MainViewModel.kt [hunk 1]
```diff
@@ -97,20 +96,21 @@ class MainViewModel : ViewModel() {
     }
 
     private suspend fun fetchData() = withContext(Dispatchers.Default) {
+        var response: retrofit2.Response<Currency>?
```
Comment (RokasMak) [inline app/src/main/java/com/ryeslim/currencyexchange/MainViewModel.kt]: Remove this line

File: app/src/main/java/com/ryeslim/currencyexchange/MainViewModel.kt [hunk 2]
```diff
@@ -97,20 +96,21 @@ class MainViewModel : ViewModel() {
...
         try {
             withContext(Dispatchers.IO) {
                 response = ServiceFactory.createRetrofitService(
```
Comment (RokasMak) [inline app/src/main/java/com/ryeslim/currencyexchange/MainViewModel.kt]: write 'val' before response

File: app/src/main/java/com/ryeslim/currencyexchange/MainViewModel.kt [hunk 3]
```diff
@@ -97,20 +96,21 @@ class MainViewModel : ViewModel() {
...
                 )
                     .getCurrencyAsync(url).await()
-            }
 
-            if (response!!.body() != null) {
-                calculateValues()
-                makeInfoMessage()
-            } else {
-                _error.postValue(Unit)
+                if (response!!.body() != null) {
```
Comment (RokasMak) [inline app/src/main/java/com/ryeslim/currencyexchange/MainViewModel.kt]: remove nullable types

**Review by RokasMak — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/ryeslim/currencyexchange/MainViewModel.kt (hunk 1 above)
Comment (r426) [reply app/src/main/java/com/ryeslim/currencyexchange/MainViewModel.kt]: removed

File: app/src/main/java/com/ryeslim/currencyexchange/MainViewModel.kt (hunk 2 above)
Comment (r426) [reply app/src/main/java/com/ryeslim/currencyexchange/MainViewModel.kt]: added

File: app/src/main/java/com/ryeslim/currencyexchange/MainViewModel.kt (hunk 3 above)
Comment (r426) [reply app/src/main/java/com/ryeslim/currencyexchange/MainViewModel.kt]: removed

File: app/src/main/java/com/ryeslim/currencyexchange/MainViewModel.kt [hunk 4]
```diff
@@ -99,18 +98,18 @@ class MainViewModel : ViewModel() {
...
-                calculateValues()
-                makeInfoMessage()
-            } else {
-                _error.postValue(Unit)
+                if (response.body() != null) {
+                    calculateValues(response.body())
+                    makeInfoMessage(response.body())
+                } else {
+                    _error.postValue(Unit)
+                }
```
Comment (RokasMak) [inline app/src/main/java/com/ryeslim/currencyexchange/MainViewModel.kt:113]: Replace this with: 
`           response.body()?.let{body ->
                    calculateValues(body)
                    makeInfoMessage(body)
                }
                if (response.body() == null) {
                    _error.postValue(Unit)
                }`

File: app/src/main/java/com/ryeslim/currencyexchange/MainViewModel.kt [hunk 5]
```diff
@@ -147,12 +146,12 @@ class MainViewModel : ViewModel() {
         thisCommission = commission.calculate(amountToConvert, numberOfOperations)
     }
 
-    private fun makeInfoMessage() {
+    private fun makeInfoMessage(responseBody: Currency?) {
```
Comment (RokasMak) [inline app/src/main/java/com/ryeslim/currencyexchange/MainViewModel.kt]: rename responseBody to currency. Remove nullable type

File: app/src/main/java/com/ryeslim/currencyexchange/MainViewModel.kt [hunk 6]
```diff
@@ -123,14 +122,14 @@ class MainViewModel : ViewModel() {
             "$amountToConvert-${(currencies[indexFrom]).currencyCode}/${(currencies[indexTo]).currencyCode}/latest"
     }
 
-    private fun calculateValues() {
+    private fun calculateValues(responseBody: Currency?) {
```
Comment (RokasMak) [inline app/src/main/java/com/ryeslim/currencyexchange/MainViewModel.kt]: rename responseBody to currency. Remove nullable type

**Review by RokasMak — CHANGES_REQUESTED** [review]

**Review by RokasMak — APPROVED** [review]

**Pull request merged**